tests/files/locations/test_locations.py

```diff
@@ -0,0 +1,163 @@
+import os
```

Comment:
What do you think about splitting the tests in a similar folder structure to the code?
E.g. to have a `test_s3.py` inside `tests/files/locations/`, etc?

Reply:
Sure